### ISSUE TITLE
`quotes`: `avoidEscape: true` doesn't run properly in the last release(s)

### ISSUE THREAD

**Issue by darksabrefr** [eslint-stylistic/eslint-stylistic#554]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Recent modifications (probably https://github.com/eslint-stylistic/eslint-stylistic/issues/542 and https://github.com/eslint-stylistic/eslint-stylistic/pull/544) seem to have changed the behavior of this rule, and I think it's a bug. It's now impossible to use backticks to wrap strings containing both single and double-quotes.

#### eslint.config.mjs
```javascript
import ESLintPluginStylistic from "@stylistic/eslint-plugin";

export default {
  plugins: {
    "@stylistic": ESLintPluginStylistic,
  },

  rules: {
    "@stylistic/quotes": [
      "warn",
      "single",
      {
        avoidEscape: true
      },
    ],
  },
};
```
#### main.js
```javascript
const a = `" ' a`;
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by manuel3108** [comment]
This seems to be related: https://github.com/eslint-stylistic/eslint-stylistic/pull/544
